### PR TITLE
Issue #75 - first cut at JUnit 5

### PR DESCRIPTION
**Pull request by alf239**
That's the initial draft; any feeedback is welcome, but for now the main goal is to see what Travis will say.

### CODE REVIEW

**Comment by apete** [comment]
Do you branch off master or develop? I prefer develop. There totally in sync now, but as a general rule.

**Comment by alf239** [comment]
Gor it regarding `develop`, will branch off it.

**Comment by alf239** [comment]
Right, `*Case` and `*Problem` are being run now.

**Comment by alf239** [comment]
Now time to tidy up and drop the rest of JUnit 3 scaffolding.

**Comment by alf239** [comment]
```
Results :
Tests run: 1442, Failures: 0, Errors: 0, Skipped: 21
```

Not bad!

**Comment by alf239** [comment]
I now have "1485 tests done, 55 ignored" in IntelliJ IDEA runner; let's see what Travis does.

**Comment by alf239** [comment]
```
Failed tests: 
  RandomNumberTest.testGamma:258->testDist2:489 expected:<1.0> but was:<0.8301302995873344>
  FundamentalScalarTest.testNegate:156->assertEqual:203 Rational expected:<8.912500543923945E-7> but was:<8.912500543923826E-7>
Tests run: 1485, Failures: 2, Errors: 0, Skipped: 55
```
1485, interesting. I, too, have 1485 now, but what's changed?

**Comment by alf239** [comment]
Oh, nevermind, was looking at the wrong numbers.

**Comment by alf239** [comment]
Judging by the comments, the "slow" tests weren't that slow after all...

Anyway, I think it's ready for a review.

File: test/org/ojalgo/constant/PrimitiveMathTest.java [hunk 1]
```diff
@@ -21,61 +21,52 @@
...
 
         final double negDbl = -0.0;
         final double posInt = 0;
         final double posDbl = 0.0;
         final double negInt = -0;
 
-        PrimitiveMathTest.compare("negDbl <-> posInt", negDbl, posInt);
-        PrimitiveMathTest.compare("negDbl <-> posDbl", negDbl, posDbl);
-        PrimitiveMathTest.compare("negDbl <-> negInt", negDbl, negInt);
+        PrimitiveMathTest.compare(negDbl, posInt);
```
Comment (alf239) [inline test/org/ojalgo/constant/PrimitiveMathTest.java]: Looks awful, I know - but then `compare` was not _using_ that string. We can roll that change back for now and fix `compare` instead.

**Comment by alf239** [comment]
Which version of Eclipse are you using — and which JUnit plugin, if any? I don't use Eclipse, so I will need to match your configuration.

**Comment by apete** [comment]
Version: Oxygen.2 Release (4.7.2)
Build id: 20171218-0600

With the  JUnit 5 test runner

I'm sure this problem will disappear when these superclasses are made public - I can fix that after I merged the PR.

**Comment by alf239** [comment]
That's weird. Kinda fixed, and you're right, it _was_ due to the package-local parents.

Still, it seems Eclipse isn't really capable of searching for tests recursively?

**Comment by alf239** [comment]
It's a known, ahem, [feature](https://stackoverflow.com/questions/413214/is-it-possible-to-run-junit-tests-from-multiple-packages-in-eclipse): you can run all of it, a single package (not including subpackages), a single class, or a single test.

Or one can do a test suite (and I did hope I retire those for good!)

**Comment by alf239** [comment]
@apete, so do you want to reinstate the old hierarchy as tags - or are you fine with the current state? From IntelliJ IDEA, slicing by package (and including subpackages) perfectly matches the old hierarchy, so I _thought_ it was redundant.

**Comment by apete** [comment]
@alf239  It's fine the way it is.

**Comment by apete** [comment]
Thanks again!

**Pull request merged**